Dockerfile.mimalloc

```diff
@@ -0,0 +1,57 @@
+# How to test changes to this file
+# docker build -f Dockerfile -t gcr.io/celo-testnet/geth:$USER .
+# To locally run that image
+# docker rm geth_container ; docker run --name geth_container gcr.io/celo-testnet/geth:$USER
+# and connect to it with
+# docker exec -t -i geth_container /bin/sh
+#
+# Once you are satisfied, build the image using
+# export COMMIT_SHA=$(git rev-parse HEAD)
+# docker build -f Dockerfile --build-arg COMMIT_SHA=$COMMIT_SHA -t gcr.io/celo-testnet/geth:$COMMIT_SHA .
+#
+# push the image to the cloud
+# docker push gcr.io/celo-testnet/geth:$COMMIT_SHA
+#
+# To use this image for testing, modify GETH_NODE_DOCKER_IMAGE_TAG in celo-monorepo/.env file
+
+# Build Geth in a stock Go builder container
+FROM golang:1.16-alpine as builder
+
+RUN apk add --no-cache make gcc musl-dev linux-headers git build-base cmake
+
+RUN cd / && \
+	git clone -b v2.0.3 --depth 1 https://github.com/microsoft/mimalloc && \
+	cd mimalloc && \
+	mkdir build && \
+	cd build && \
+	cmake .. && \
+	make -j$(nproc) && \
+	make install
+
+ADD . /go-ethereum
+RUN cd /go-ethereum && make geth-musl
+
+# Pull Geth into a second stage deploy alpine container
+FROM alpine:latest
+ARG COMMIT_SHA
+
+# Copy over compiled libmimalloc
+COPY --from=builder /mimalloc/build/*.so.* /lib
+RUN ln -s /lib/libmimalloc.so.* /lib/libmimalloc.so
+ENV LD_PRELOAD=/lib/libmimalloc.so
```

Comment:
if this `LD_PRELOAD` env variable is not set, then geth will still run with the default musl memory allocator